ray_beam_runner/portability/state.py

```diff
@@ -91,7 +91,8 @@ def clear(self, bundle_id: str, state_key: str):
 class RayStateManager(sdk_worker.StateHandler):
     def __init__(self, state_actor: Optional[_ActorStateManager] = None):
         self._state_actor = state_actor or _ActorStateManager.remote()
-        self._instruction_id: Optional[str] = None
+        # TODO(iasoon): how do we assign the correct instruction id for side inputs?
```

Comment:
@pabloem Could you shed some light on this? We are scoping state access by the owning instruction_id, but I don't think we have access to the instruction_id that would need a side input at the time that we commit the side inputs to the state manager.

Reply:
Looks like we could not find the instruction_id because we assign it ourselves :)

to fix this, I assigned an uid to use for its processing when we construct the `RayRunnerExecutionContext`. Is it correct that there should be an one-to-one mapping between stages and bundle ids?